external/readme.md

```diff
@@ -0,0 +1 @@
+directory for git submodules
```

Comment:
todo: add a list of submodules with a brief (or detailed, if it feels necessary, e.g. "why this instead of that?" if you want) description of each and why they're included in the project / what purpose they serve it.